BlackMarket/app/Routing/StackRoutesRegistrationModifier.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  StackRoutesRegistrationModifier.swift
+//  BlackMarket
+//
+//  Created by Tarek Radovan on 17/02/2023.
+//
+
+import Foundation
+import Combine
+import SwiftUI
+
+/// Each view registers its routes via this ViewModifier.
+struct StackRoutesRegistrationModifier<R: Router>: ViewModifier {
+
+  private let router: R
+
+  init(router: R) {
+    self.router = router
+  }
+
+  func body(content: Content) -> some View {
+    if #available(iOS 16.0, *) {
```

Comment:
I made this in the playground for testing purposes, we should declare this entire type to be available in the iOS version we want or find a fallback solution?
Otherwise, this modifier can be used but does nothing in earlier versions

Reply:
I would put a base version of 16.0 on the app, otherwise we would have to look for options for lower versions and it would be quite ugly and non-performing code that would be quite limited compared to NavigationStack

Reply:
The problem with managing NavigationLink for lower versions is that we would have to add it everywhere, which would also require us to add comparisons between iOS 16 or lower. This would result in a disaster and add a lot of odd behaviors due to the difference in how NavigationLink and NavigationStack work